### PR TITLE
Bluetooth: Controller: Fix spurious ISO Sync receiver stall

### LINKED ISSUE #80734: tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_12.sh failing in main
**Describe the bug**
One BT audio CAP Broadcast AC test is failing in main
tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_12.sh 
in its "Execute_AC_12 48_4_1" step

**To Reproduce**
Make sure you have bsim installed and are ready to build targeting the nrf52_bsim
1. tests/bsim/bluetooth/audio/compile.sh
2. tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_12.sh

**Expected behavior**
No test failures

**Impact**
Failures in CI, blocking PRs

**Logs and console output**
```
...
d_00: @00:00:58.014617  [5800]: Stream 0x84d4240 sent with seq_num 5800
d_01: @00:00:58.077226  Received BASE with 1 subgroups from broadcast sink 0x854a3a0
d_01: @00:00:58.077226  Metadata updated
d_01: @00:00:58.077226  type 2 len 2
d_01: @00:00:58.617227  Received BASE with 1 subgroups from broadcast sink 0x854a3a0
d_01: @00:00:58.617227  Metadata updated
d_01: @00:00:58.617227  type 2 len 2
d_00: @00:00:59.014617  [5900]: Stream 0x84d4240 sent with seq_num 5900
d_01: @00:00:59.157227  Received BASE with 1 subgroups from broadcast sink 0x854a3a0
d_01: @00:00:59.157227  Metadata updated
d_01: @00:00:59.157227  type 2 len 2
d_01: @00:00:59.697227  Received BASE with 1 subgroups from broadcast sink 0x854a3a0
d_01: @00:00:59.697227  Metadata updated
d_01: @00:00:59.697227  type 2 len 2
d_00: @00:01:00.000722  TESTCASE NOT PASSED at exit (test return (1) indicates it was still in progress)
d_01: @00:01:00.008038  TESTCASE NOT PASSED at exit (test return (1) indicates it was still in progress)
```

**Environment (please complete the following information):**

 - OS: Ubuntu 24.04 
 - Toolchain (e.g Zephyr SDK, ...): Host gcc 14
 - Commit SHA or Version used: main after f970b066d210af56ab6159b2c22e825230c212b9

**Additional context**
Failure started after https://github.com/zephyrproject-rtos/zephyr/pull/80228 got merged
f970b066d210af56ab6159b2c22e825230c212b9

### PR DESCRIPTION
**Pull request by cvinayak**
Fix spurious ISO Sync Receiver stall due to uninitialised
value accessed due to regression introduced by
commit https://github.com/cvinayak/zephyr/commit/64faceea7270dd2299c3602b766dcfb935b7fc84 ("Bluetooth: controller: Stop Sync ISO
ticker when establishment fails").

Align audio test Controller Kconfig value same as used with nRF53bsim.

Fixes #80734.

### CODE REVIEW

**Review by aescolar — APPROVED** [review]
Thank you

**Comment by aescolar** [comment]
@dkalowsk @mmahadevan108 It would be very nice to get this for 4.0, otherwise we should disable the failing test case (it would be quite bad to release 4.0 with a failing BT testcase as that would prevent backporting any BT related fix)

**Comment by Thalley** [comment]
Will try to test this with my PR that saw a similar problem

File: tests/bsim/bluetooth/audio/overlay-bt_ll_sw_split.conf
Comment (Thalley) [inline tests/bsim/bluetooth/audio/overlay-bt_ll_sw_split.conf:1]: Could we instead just use the nRF53 BSIM .conf file?

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:1]: I had a feeling it was related to uninitialzed data, but valgrind didn't show anything for me. How did you find this?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:1]: Had to debug inside the lower link layer to check when the reception stopped, there were calls to `ticker_stop` and lead to the caller which used this uninitialized value (recently added struct member). Since the value is from inside the Rx buffer node allocated from pool, the overall content get assigned something sometime before when receiving other packets.

File: tests/bsim/bluetooth/audio/overlay-bt_ll_sw_split.conf
Comment (cvinayak) [reply tests/bsim/bluetooth/audio/overlay-bt_ll_sw_split.conf:1]: Will investigate as part of enabling some high reliability CAP tests. Reusing the target conf file is challenging as high reliability test need lot more tx and rx buffers to handle worst case retransmissions. 

Trying out fixes and Kconfig changes here: https://github.com/zephyrproject-rtos/zephyr/pull/80788

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:1285]: This value isn't assigned after line 1252 - Is that intentional?

**Comment by Thalley** [comment]
I think this fixes the stalling issue (passes those tests locally), but it will fail other tests from my PR https://github.com/zephyrproject-rtos/zephyr/pull/80691:

```
======== Running CAP AC_13 with 441_1_1 =========
...
d_01: @00:00:01.392685  [0]: Incoming audio on stream 0x84da700 len 97 and ts 1376271 flags 0x09
d_01: @00:00:01.392685  [0]: Incoming audio on stream 0x84da700 len 97 and ts 1376271
d_01: @00:00:01.392685  [0]: Incoming audio on stream 0x84da744 len 97 and ts 1383748 flags 0x09
d_01: @00:00:01.392685  [0]: Incoming audio on stream 0x84da744 len 97 and ts 1383748
d_01: @00:00:01.400187  [1]: Incoming audio on stream 0x84da700 len 0 and ts 1384434 flags 0x0c
d_01: @00:00:01.400187 ERROR: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/audio/src/cap_acceptor_test.c:321): ISO receive lost
```

This is a broadcast test, and the host always enqueues 2 SDUs per stream in the controller. 

I think it's OK to merge, as it fixes the issues it claims, but it does not solve all issues. Will spend some more time tomorrow verifying that the host correctly enqueues the number of SDUs it expects.

**Review by Thalley — APPROVED** [review]

**Comment by dkalowsk** [comment]
> @dkalowsk @mmahadevan108 It would be very nice to get this for 4.0, otherwise we should disable the failing test case (it would be quite bad to release 4.0 with a failing BT testcase as that would prevent backporting any other BT related fix)

@aescolar thanks for bringing it to attention.   It does have the associated Issue to fix so as far as I can tell this will work on RC2 provided maintainers are good with it (and it looks like Thally is)

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:1285]: Yes, intentional. Only relevant structure members are assigned for each type being passed around.

Controller has large Rx data flow (in struct node_rx) overheads (two parts: fixed overhead of unions, and configurable PDU length payload). Generally, not all members of the overhead part (respective to the type of rx node) are assigned.

When adding new struct member, typically all sources of the rx node type in the data flow need to be inspected :-(

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:1285]: On second thought i need to investigate further today...https://github.com/zephyrproject-rtos/zephyr/pull/80788  pass locally, but fails in CI :-(

**Comment by cvinayak** [comment]
@aescolar @Thalley any new PR fails CI with different audio tests failing for same commit HEAD with re-runs: https://github.com/zephyrproject-rtos/zephyr/actions/runs/11666290372?pr=80788
May be many tests fail in CI (that do not fail locally or vice versa), but CI stops on first failure of many running in parallel?

Comparing the bsim timestamps between CI and local run (local native posix toolchain), they do not appear same and i suspect timing change in bsim run on CI too for re-runs.

Timing compare here: https://github.com/zephyrproject-rtos/zephyr/pull/80788#issuecomment-2454067810

**Comment by cvinayak** [comment]
> I have only seen it happen on the nRF52 BSIM, and have not been able to reproduce with the nRF53 BSIM nor on target.

Refer to https://github.com/zephyrproject-rtos/zephyr/issues/80734#issuecomment-2451823771

The conf change in the commit in this PR is to address the inconsistency between nrf52 bsim and nrf53 bsim testing.

**Comment by cvinayak** [comment]
@aescolar  I do not like that you have independently opened two new PRs (#80894, and ##80896) and marked them as hotfixes while this PR is made to wait for the merge timeout.

**Comment by cvinayak** [comment]
Replaced by #80894.

**Pull request closed without merging**
